gr-digital/lib/costas_loop_cc_impl.h

```diff
@@ -35,28 +35,59 @@ class costas_loop_cc_impl : public costas_loop_cc
     const unsigned int d_order;
     float d_error;
     float d_noise;
+    bool d_use_snr;
 
     /*! \brief the phase detector circuit for 8th-order PSK loops.
      *
      *  \param sample complex sample
      *  \return the phase error
      */
-    float phase_detector_8(gr_complex sample) const; // for 8PSK
+    float phase_detector_8(gr_complex sample) const // for 8PSK
+    {
+        /* This technique splits the 8PSK constellation into 2 squashed
+   QPSK constellations, one when I is larger than Q and one
+   where Q is larger than I. The error is then calculated
+   proportionally to these squashed constellations by the const
+   K = sqrt(2)-1.
+
+   The signal magnitude must be > 1 or K will incorrectly bias
+   the error value.
+
+   Ref: Z. Huang, Z. Yi, M. Zhang, K. Wang, "8PSK demodulation for
+   new generation DVB-S2", IEEE Proc. Int. Conf. Communications,
+   Circuits and Systems, Vol. 2, pp. 1447 - 1450, 2004.
```

Comment:
Fix indention.

Reply:
ping on this change

Reply:
If you are wondering which changes Micheal is referring to (they seem to be buried by now):
https://github.com/gnuradio/gnuradio/pull/3076#discussion_r370948883